Gems/Atom/Utils/Code/Source/AssetCollectionAsyncLoader.cpp

```diff
@@ -220,10 +222,10 @@ namespace AZ
             m_assetIdStrToAssetPath[assetIdStr] = assetPath;
         }
 
-        Data::AssetBus::MultiHandler::BusConnect(assetId);
         // Kick off asset loading.
-        auto asset = Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::PreLoad);
-        asset.BlockUntilLoadComplete();
+        auto asset = Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::QueueLoad);
+        m_notReadyAssets[assetPath] = asset;
```

Comment:
In other functions you have protected m_notReadyAssets with m_mutex, should it be protected here too?

Reply:
You are right. Fixed.